clippy_lints/src/disallowed_type.rs

```diff
@@ -65,6 +65,12 @@ impl_lint_pass!(DisallowedType => [DISALLOWED_TYPE]);
 impl<'tcx> LateLintPass<'tcx> for DisallowedType {
     fn check_crate(&mut self, cx: &LateContext<'_>, _: &Crate<'_>) {
         for path in &self.disallowed {
+            if path.len() == 1 {
+                if let Some(ty) = PrimTy::from_name(path[0]) {
+                    self.prim_ty.insert(ty);
+                    continue;
+                }
+            }
```

Comment:
This seemed the most efficient and direct way to do this, if you were thinking something else let me know!

Reply:
Unfortunately, I can't use `path_to_res` for primitive types unless they are of the form `crate::[mod]::name` that's why I was asking about `Symbol` to `DefId` (although looking further at how rustc handles primitives they seem to be a separate thing having their `DefId` defined in a macro via `LangItems` or something like that).

Reply:
Ahh do you mean add the above code to the `path_to_res` function?